ecommerce-ui/src/components/ItemView.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import PropertyCard from './PropertyCard';
+import itemData from './airbnbs.json';
+import Header from './Header';
+
+class ItemView extends React.Component {
+  state = {
+    selectedProperty: [{key: 1, title:"", cost:0}],
+    stateKey: 1
+  }
+
+  addRental = (title) => {
+
+    itemData.map(items => {
+
+      if (title === items.title){
+        const newKey = this.state.stateKey + 1;
+        const selectedProp = {title:items.title, cost:items.payment.cost, key: newKey}
+        this.setState(prevState => {
```

Comment:
Interesting use of an iterator to create a unique key! I am still not sure of best practice though as Brian has mentioned in production you will usually be using a UID tied to an object.